File: examples/OutermindUI/src/OutermindUI.WebAssembly/wwwroot/outermindJSInterop.js
window.OutermindJSFunctions = {
    alert: function(message) {
        alert(message);
        return true;
    },
    capturePointer: function (elementId, pointerId) {
        const element = document.getElementById(elementId)

        if (element !== null) {
            element.setPointerCapture(pointerId);
            return true;
        }

        return false;
    },
    releasePointer: function (elementId, pointerId) {
        const element = document.getElementById(elementId)

        if (element !== null) {
            element.releasePointerCapture(pointerId);
            return true;
        }

        return false;
    }
}

File: examples/OutermindUI/src/OutermindUI.WebAssembly/OutermindJSInterop.cs
using Microsoft.JSInterop;

namespace OutermindUI;

internal static class OutermindJSInterop
{
    internal static ValueTask<bool> Alert(this IJSRuntime jsRuntime, string message) =>
        jsRuntime.InvokeAsync<bool>("OutermindJSFunctions.alert", message);

    internal static ValueTask<bool> CapturePointer(this IJSRuntime jsRuntime, string elementId, long pointerId) =>
        jsRuntime.InvokeAsync<bool>("OutermindJSFunctions.capturePointer", elementId, pointerId);

    internal static ValueTask<bool> ReleasePointer(this IJSRuntime jsRuntime, string elementId, long pointerId) =>
        jsRuntime.InvokeAsync<bool>("OutermindJSFunctions.releasePointer", elementId, pointerId);
}
